packages/block-editor/src/components/block-settings-menu-controls/index.js

```diff
@@ -37,12 +37,16 @@ const BlockSettingsMenuControlsSlot = ( { fillProps, clientIds = null } ) => {
 	return (
 		<Slot fillProps={ { ...fillProps, selectedBlocks } }>
 			{ ( fills ) => {
-				return (
-					<MenuGroup>
-						{ fills }
-						<ConvertToGroupButton onClose={ fillProps?.onClose } />
-					</MenuGroup>
-				);
+				if ( fills?.length > 0 ) {
+					return (
+						<MenuGroup>
+							{ fills }
+							<ConvertToGroupButton
```

Comment:
Question, this Convert to Group button is now wrapped in the same `fills?.length` condition. Is that appropriate?

Reply:
I missed that to be honest. It will require some tweaks. @ntsekouras, can you help? This convert to buttons part is also optional 🙃

Reply:
I figured it might be tricky 🙈 — that's why I kept this in a draft state. But hopefully this will get the ball rolling, because the empty menu group is not great, I imagine for screenreaders either.

Reply:
It might be fine with screenreaders since there are no focusable items inside the group. I speculate though. It's an issue that can be and should be addressed 😄

Reply:
I created a new hook for this here: https://github.com/WordPress/gutenberg/pull/28486/commits/7429243488ec413ecd927df378f8b894fe78f4d1